packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/resources/knowledgemiddleware/KnowledgeResource.java

```diff
@@ -106,16 +106,11 @@ public Model postLaTeXToAMR(
 	@Path("/code-to-amr")
 	@Consumes(MediaType.APPLICATION_JSON)
 	public ExtractionResponse postCodeToAMR(
-		String artifactId,
-		String name,
-		String description
+		@QueryParam("code_id") String codeId,
+		@QueryParam("name") String name,
+		@QueryParam("description") String description
 	) {
-		// Fetch the related artifact to fill potential missing name and description
-		final Artifact artifact = artifactProxy.getArtifact(artifactId);
-		if (name == null) {	name = artifact.getName(); }
-		if (description == null) { description = artifact.getDescription();	}
-
-		return knowledgeMiddlewareProxy.postCodeToAMR(artifactId, name, description);
+		return knowledgeMiddlewareProxy.postCodeToAMR(codeId, name, description);
```

Comment:
[It returns more](https://github.com/DARPA-ASKEM/knowledge-middleware/blob/main/worker/operations.py#L509-L515) than just a `Model`.
This would be useful to maybe display the model directly by open it instead of returning the AMR?

Reply:
Not sure what you mean. The model already gets saved in tds as a result of calling code-to-amr?

Reply:
Right now we use the response to display the AMR, but we still don't open it, there is a `createModel` call made after this.